src/app/features/location/address-gps-location/address-gps-location.component.ts

```diff
@@ -0,0 +1,57 @@
+import { Component } from "@angular/core";
+import { Logging } from "app/core/logging/logging.service";
+import { GpsService } from "../gps.service";
+import { MatTooltip } from "@angular/material/tooltip";
+import { FaIconComponent } from "@fortawesome/angular-fontawesome";
+import { MatProgressSpinnerModule } from "@angular/material/progress-spinner";
+import { MatIconButton } from "@angular/material/button";
+import { NgIf } from "@angular/common";
+import { MapPopupComponent } from "../map-popup/map-popup.component";
+import { Coordinates } from "../coordinates";
+import { AlertService } from "app/core/alerts/alert.service";
+
+@Component({
+  selector: "app-address-gps-location",
+  standalone: true,
+  imports: [
+    MatTooltip,
+    FaIconComponent,
+    MatProgressSpinnerModule,
+    NgIf,
+    MatTooltip,
+    MatIconButton,
+  ],
+  templateUrl: "./address-gps-location.component.html",
+  styleUrl: "./address-gps-location.component.scss",
+})
+export class AddressGpsLocationComponent {
+  constructor(
+    private gpsService: GpsService,
+    private mapPopupComponent: MapPopupComponent,
+    private alertService: AlertService,
+  ) {}
+  location: Coordinates;
+  public gpsLoading = false;
+
+  async updateLocationFromGps() {
+    this.gpsLoading = true;
+    try {
+      const location = await this.gpsService.getGpsLocationCoordinates();
+      if (location) {
+        this.location = {
+          lat: location.latitude,
+          lon: location.longitude,
+        };
+        await this.mapPopupComponent.mapClicked(this.location);
+        this.alertService.addInfo("Selected address based on GPS.");
+      }
+    } catch (error) {
+      Logging.error(error);
+      this.alertService.addInfo(
+        "Location permission denied. Please enable it in your device settings.",
```

Comment:
Are you sure this always fails due to permissions? Or can there be other causes and we are possibly sending the user in a wrong direction here?

Reply:
@sleidig I reviewed this scenario and noticed that when the user denies the permission, an alert message is shown. The same alert message also appears if any code within the try block fails.

Reply:
So that means the most likely explanation is the permissions but it could also be any other kind of error. Then we should change that message to users a bit

```suggestion
        $localize`Failed to access device location. Please check if location permission is enabled in your device settings`,
```